lib/logstash/inputs/http_poller.rb

```diff
@@ -55,9 +55,12 @@ def register
     setup_requests!
   end
 
+  # @overload
   def stop
-    Stud.stop!(@interval_thread) if @interval_thread
-    @scheduler.stop if @scheduler
+    if @scheduler && @scheduler.thread && @scheduler.thread.status
+      @scheduler.shutdown # on newer Rufus (3.8) this joins on the scheduler thread
+    end
+    # TODO implement client.close as we as releasing it's pooled resources!
```

Comment:
I've a doubt on this, if in the `.travis.yml` and `Gemfile` we pin to Rufus `3.0.9` because LS 7.x pins that version, is this code `7.x` proof?
Do this change behaves the same way also with Rufus `3.0.9` ?

Reply:
the missing `client.close` is yet another completely separate issue (basically a plugin leak on reload) - stumbled upon it while trying to investigate the CI failure - please ignore the TODO for now if it's confusing.

> I've a doubt on this, if in the `.travis.yml` and `Gemfile` we pin to Rufus 3.0.9 because LS 7.x pins that version, is this code 7.x proof?

it's proof that tests run (on CI) against latest rufus-scheduler as well as 3.0.9. LS 7.x for now used 3.0.9 but in theory (when the plugin unpinning from https://github.com/elastic/logstash/issues/12931 is complete) with the right scenario of multpile `bin/logstash-plugin update ...` the lock could be released and rufus could be updated.

previous version had `@scheduler.shutdown if @scheduler`, in theory this could still fail if the thread did not start or the thread died - in those cases we can not trigger a "safe" `scheduler.shutdown`. but this isn't smt we're seeing and I can revert the code to be the same as before if you have doubts (that change isn't that important atm).

Reply:
I'm fine with this, thanks for clarification.